### ISSUE TITLE
微调训练无法占满GPU

### ISSUE THREAD

**Issue by wuhuanon** [ymcui/Chinese-LLaMA-Alpaca-2#204]
### 提交前必须检查以下项目

- [X] 请确保使用的是仓库最新代码（git pull），一些问题已被解决和修复。
- [X] 我已阅读[项目文档](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki)和[FAQ章节](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki/常见问题)并且已在Issue中对问题进行了搜索，没有找到相似问题和解决方案。
- [X] 第三方插件问题：例如[llama.cpp](https://github.com/ggerganov/llama.cpp)、[LangChain](https://github.com/hwchase17/langchain)、[text-generation-webui](https://github.com/oobabooga/text-generation-webui)等，同时建议到对应的项目中查找解决方案。

### 问题类型

模型训练与精调

### 基础模型

Alpaca-2-7B

### 操作系统

Linux

### 详细描述问题

# 运行脚本前请仔细阅读wiki(https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki/sft_scripts_zh)
# Read the wiki(https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki/sft_scripts_zh) carefully before running the script
lr=1e-4
lora_rank=64
lora_alpha=128
lora_trainable="q_proj,v_proj,k_proj,o_proj,gate_proj,down_proj,up_proj"
modules_to_save="embed_tokens,lm_head"
lora_dropout=0.05

pretrained_model=/model/chinese-alpaca-llama-2
chinese_tokenizer_path=/model/chinese-alpaca-llama-2
dataset_dir=Chinese-LLaMA-Alpaca-2-main/data/default
peft_model=/model/chinese-alpaca-2-lora-7b
per_device_train_batch_size=1
per_device_eval_batch_size=1
gradient_accumulation_steps=8
max_seq_length=1024
output_dir=/model/sft
validation_file=/data/default/alpaca_data_zh_51k.json

deepspeed_config_file=ds_zero2_no_offload.json

torchrun --nnodes 1 --nproc_per_node 1 run_clm_sft_with_peft.py \
    --model_name_or_path ${pretrained_model} \
    --tokenizer_name_or_path ${chinese_tokenizer_path} \
    --dataset_dir ${dataset_dir} \
    --per_device_train_batch_size ${per_device_train_batch_size} \
    --per_device_eval_batch_size ${per_device_eval_batch_size} \
    --do_train \
    --do_eval \
    --seed 100 \
    --fp16 \
    --num_train_epochs 1 \
    --lr_scheduler_type cosine \
    --learning_rate ${lr} \
    --warmup_ratio 0.03 \
    --weight_decay 0 \
    --logging_strategy steps \
    --logging_steps 10 \
    --save_strategy steps \
    --save_total_limit 3 \
    --evaluation_strategy steps \
    --eval_steps 100 \
    --save_steps 200 \
    --gradient_accumulation_steps ${gradient_accumulation_steps} \
    --preprocessing_num_workers 8 \
    --max_seq_length ${max_seq_length} \
    --output_dir ${output_dir} \
    --overwrite_output_dir \
    --ddp_timeout 30000 \
    --logging_first_step True \
    --lora_rank ${lora_rank} \
    --lora_alpha ${lora_alpha} \
    --trainable ${lora_trainable} \
    --lora_dropout ${lora_dropout} \
    --torch_dtype float16 \
    --validation_file ${validation_file}
我用的3090跑的，GPU只占到60%左右

### 依赖情况（代码类问题务必提供）

absl-py                        1.1.0
accelerate                     0.22.0
aiofiles                       23.2.1
aiohttp                        3.8.5
aiosignal                      1.3.1
altair                         5.0.1
annotated-types                0.5.0
anyio                          3.6.1
argon2-cffi                    21.3.0
argon2-cffi-bindings           21.2.0
asttokens                      2.0.5
async-timeout                  4.0.3
attrs                          21.4.0
Babel                          2.10.3
backcall                       0.2.0
beautifulsoup4                 4.11.1
bitsandbytes                   0.41.0
bleach                         5.0.0
brotlipy                       0.7.0
cachetools                     5.2.0
certifi                        2021.5.30
cffi                           1.14.6
chardet                        4.0.0
charset-normalizer             3.2.0
click                          8.1.7
cmake                          3.27.2
conda                          4.10.3
conda-package-handling         1.7.3
cryptography                   3.4.7
cycler                         0.11.0
datasets                       2.14.4
debugpy                        1.6.0
decorator                      5.1.1
deepspeed                      0.10.1
defusedxml                     0.7.1
dill                           0.3.7
entrypoints                    0.4
executing                      0.8.3
fastapi                        0.103.0
fastjsonschema                 2.15.3
ffmpy                          0.3.1
filelock                       3.12.2
fonttools                      4.33.3
frozenlist                     1.4.0
fsspec                         2023.6.0
google-auth                    2.8.0
google-auth-oauthlib           0.4.6
gradio                         3.41.2
gradio-client                  0.5.0
grpcio                         1.46.3
h11                            0.14.0
hjson                          3.1.0
httpcore                       0.17.3
httpx                          0.24.1
huggingface-hub                0.16.4
idna                           2.10
importlib-metadata             4.11.4
importlib-resources            5.8.0
ipykernel                      6.15.0
ipython                        8.4.0
ipython-genutils               0.2.0
ipywidgets                     7.7.0
jedi                           0.18.1
Jinja2                         3.1.2
joblib                         1.3.2
json5                          0.9.8
jsonschema                     4.6.0
jupyter-client                 7.3.4
jupyter-core                   4.10.0
jupyter-server                 1.17.1
jupyterlab                     3.4.3
jupyterlab-language-pack-zh-CN 3.4.post1
jupyterlab-pygments            0.2.2
jupyterlab-server              2.14.0
jupyterlab-widgets             1.1.0
kiwisolver                     1.4.3
linkify-it-py                  2.0.2
lit                            16.0.6
Markdown                       3.3.7
markdown-it-py                 2.2.0
MarkupSafe                     2.1.1
matplotlib                     3.5.2
matplotlib-inline              0.1.3
mdit-py-plugins                0.3.3
mdurl                          0.1.2
mistune                        0.8.4
mpmath                         1.3.0
multidict                      6.0.4
multiprocess                   0.70.15
nbclassic                      0.3.7
nbclient                       0.6.4
nbconvert                      6.5.0
nbformat                       5.4.0
nest-asyncio                   1.5.5
networkx                       3.1
ninja                          1.11.1
notebook                       6.4.12
notebook-shim                  0.1.0
numpy                          1.22.4
oauthlib                       3.2.0
orjson                         3.9.5
packaging                      21.3
pandas                         2.0.3
pandocfilters                  1.5.0
parso                          0.8.3
peft                           0.3.0.dev0
pexpect                        4.8.0
pickleshare                    0.7.5
Pillow                         9.1.1
pip                            21.1.3
prometheus-client              0.14.1
prompt-toolkit                 3.0.29
protobuf                       3.19.4
psutil                         5.9.1
ptyprocess                     0.7.0
pure-eval                      0.2.2
py-cpuinfo                     9.0.0
pyarrow                        13.0.0
pyasn1                         0.4.8
pyasn1-modules                 0.2.8
pycosat                        0.6.3
pycparser                      2.20
pydantic                       1.10.12
pydantic-core                  2.6.3
pydub                          0.25.1
Pygments                       2.12.0
pyOpenSSL                      20.0.1
pyparsing                      3.0.9
pyrsistent                     0.18.1
PySocks                        1.7.1
python-dateutil                2.8.2
python-multipart               0.0.6
pytz                           2022.1
PyYAML                         6.0.1
pyzmq                          23.2.0
regex                          2023.8.8
requests                       2.25.1
requests-oauthlib              1.3.1
rsa                            4.8
ruamel-yaml-conda              0.15.100
safetensors                    0.3.3
scikit-learn                   1.3.0
scipy                          1.10.1
semantic-version               2.10.0
Send2Trash                     1.8.0
sentencepiece                  0.1.97
setuptools                     52.0.0.post20210125
six                            1.16.0
sniffio                        1.2.0
soupsieve                      2.3.2.post1
stack-data                     0.3.0
starlette                      0.27.0
supervisor                     4.2.4
sympy                          1.12
tensorboard                    2.9.1
tensorboard-data-server        0.6.1
tensorboard-plugin-wit         1.8.1
terminado                      0.15.0
threadpoolctl                  3.2.0
tinycss2                       1.1.1
tokenizers                     0.13.3
toolz                          0.12.0
torch                          2.0.1+cu118
torchvision                    0.12.0+cu113
tornado                        6.1
tqdm                           4.66.1
traitlets                      5.3.0
transformers                   4.31.0
triton                         2.0.0
typing-extensions              4.7.1
tzdata                         2023.3
uc-micro-py                    1.0.2
urllib3                        1.26.6
uvicorn                        0.23.2
wcwidth                        0.2.5
webencodings                   0.5.1
websocket-client               1.3.3
websockets                     11.0.3
Werkzeug                       2.1.2
wheel                          0.36.2
widgetsnbextension             3.6.0
xxhash                         3.3.0
yarl                           1.9.2
zipp                           3.8.0

### 运行日志或截图

```
# 请在此处粘贴运行日志（请粘贴在本代码块里）
```

**Comment by bugface** [comment]
per_device_train_batch_size=2 or 4?
